ome/provider.go

```diff
@@ -219,6 +219,7 @@ func (p *omeProvider) DataSources(ctx context.Context) []func() datasource.DataS
 		NewVlanNetworkDataSource,
```

Comment:
```
@@ -219,6 +219,7 @@ func (p *omeProvider) DataSources(ctx context.Context) []func() datasource.DataS
 	fmt.Println("DataSources:", p.DataSources)
 	return p.DataSources
 }

```
The only addition in this diff is the `NewAppCertDataSource` in the `DataSources` function. Overall, the code looks clean, but I have a few suggestions to improve it:

1. **Naming Convention**: The function name `DataSources` is not in camelCase, which is the standard naming convention for Go. It should be renamed to `dataSources`.
2. **Functional Return Type**: The function `dataSources` is returning a slice of functions, which might not be the most efficient way. It would be better to return the slice of strings directly, or define a custom `DataSource` struct to hold relevant information.
3. **Code Organization**: The `dataSources` function seems to be listing various data sources. It might be more organized to separate these data sources into a separate package or struct for better maintainability and readability.

Please note that without access to the full codebase, it's challenging to provide more specific suggestions. However, the above points should be considered as general improvements.

**Code Snippet:**
```go
type DataSource struct {
    NewVlanNetworkDataSource
    NewConfigurationReportDataSource
    NewDeviceDatasource
    NewAppCertDataSource
}

func (p *omeProvider) dataSources(ctx context.Context) []string {
    return []string{"NewVlanNetworkDataSource", "NewConfigurationReportDataSource", "NewDeviceDatasource", "NewAppCertDataSource"}
}
```
By implementing these suggestions, the code will be more readable, maintainable, and adherent to the Go language conventions.